.github/workflows/csqa.yml

```diff
@@ -105,3 +105,35 @@ jobs:
 
       - name: Run Static Analysis
         run: composer static-analysis
+
+  perf-guard:
+    name: "Basic Performance Guard"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "8.1"
+          coverage: none
+
+      # Install dependencies and handle caching in one go.
+      # Dependencies need to be installed to make sure the PHPUnit classes are recognized.
+      # @link https://github.com/marketplace/actions/install-php-dependencies-with-composer
+      - name: Install Composer dependencies
+        uses: "ramsey/composer-install@v3"
+        with:
+          # Bust the cache at least once a month - output format: YYYY-MM.
+          custom-cache-suffix: $(date -u "+%Y-%m")
+
+      - name: Download performance test fixture
+        run: wget https://raw.githubusercontent.com/WordPress/WordPress/refs/heads/master/wp-includes/PHPMailer/PHPMailer.php
```

Comment:
Suggestion: maybe use the PHPMailer version from the original repo ? Instead of the ripped off copy WP contains ? Might also be polite to ask @Synchro if he's okay with PHPMailer being used for these tests ?

Reply:
I basically just need a big and relatively complex PHP file. I plucked this out of the WP repo since I figured it's OSS but I'd be happy to use one from the original repo (I didn't realize this was duplicated). I'm not sure how to go about picking a good target, but I probably want to pin whatever file I use on a version for consistency. 🤔

Reply:
https://github.com/PHPMailer/PHPMailer/ - and you should be able to pin the download on a tagged version ;-)

Reply:
This is fine by me. Happy to provide a guinea pig :)